File: cpp/Backjoon.10799/main.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main(void){
    int ans = 0;
    string input = "";
    stack<char> st;

    cin >> input;

    for(int i = 0; i < input.size(); i++){
        if(input[i] == '(' && input[i+1] == ')'){
            //cut
            ans += st.size();
            i = i+1;
        }else if(input[i] == '('){
            st.push(input[i]);
        }else if(input[i] == ')'){
            st.pop();
            ans += 1;
        }
        
    }

    cout << ans << endl;
    
    return 0;
}

File: cpp/Backjoon.10799/main.py
class stack:
    def __init__(self):
        self.items = []
    def push(self, item):
        self.items.append(item)
    def pop(self):
        return self.items.pop()
    def isEmpty(self):
        return not self.items
    def size(self):
        return len(self.items)

st = stack()
ans = 0
str = input()

for i in range(0, len(str)):
    if str[i] == '(' and str[i+1] == ')':
        ans += st.size()
    elif str[i] == '(':
        st.push(str[i])
    elif str[i] == ')' and str[i-1] == '(':
        continue
    elif st.size() > 0 and str[i] == ')':
        st.pop()
        ans += 1

print(ans)
